src/components/ui/QuestionBox/QuestionBox.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from "react";
+import { StyledQuestionBox } from "./QuestionBox.style";
+import dummyData from "../../../api/dummyData";
+import { AnswerButton } from "../QuestionBox/QuestionBox.style";
+
+const QuestionBox = () => {
+  const [answerColors, setAnswerColors] = useState(
+    Array(4).fill("var(--color-for-pre-answer)"),
+  );
+  const [isAnswered, setIsAnswered] = useState(false);
+
+  const questionData = dummyData[0].questions[0];
+
+  const handleAnswerClick = (index) => {
+    if (isAnswered) return; // If already answered, do nothing
+
+    const updatedColors = [...answerColors];
+
+    if (index === questionData.correctAnswer) {
+      updatedColors[index] = "var(--color-for-correct-answer)";
+    } else {
+      updatedColors[index] = "var(--color-for-wrong-answer)";
+      updatedColors[questionData.correctAnswer] = "var(--color-for-correct-answer)";
+    }
+
+    setAnswerColors(updatedColors);
+    setIsAnswered(true);
+  };
+
+  return (
+    <StyledQuestionBox>
```

Comment:
One other convention which I didnt share with you is that for styled components it is easier to read and understand that they are styled components if they have a convention prefix as S
you do it as follows:
import * as S from "./foo.styles.js
<S.QuestionBox>
Please see the Button.jsx for reference

Reply:
like that in the QuestionBox.style .js :
![image](https://github.com/user-attachments/assets/7b333b24-7685-42df-b385-b2d9f1945d9f)

QuestionBox.jsx :
![image](https://github.com/user-attachments/assets/2d1628f9-fad3-4ee0-bbbe-9a8664d0b43e)
![image](https://github.com/user-attachments/assets/5cde7260-80cd-420d-b312-4f82b33b45c2)